dotcom-rendering/src/web/components/ShowMore.importable.tsx

```diff
@@ -0,0 +1,88 @@
+import { css } from '@emotion/react';
+import { from, space } from '@guardian/source-foundations';
+import { Button, SvgMinus, SvgPlus } from '@guardian/source-react-components';
+import { useState } from 'react';
+import { useApi } from '../lib/useApi';
+
+function decideButtonText({
+	showMore,
+	loading,
+	displayName,
+}: {
+	showMore: boolean;
+	loading: boolean;
+	displayName: string;
+}) {
+	if (showMore && loading) return 'Loading';
+	if (showMore) return 'Less';
+	return `More ${displayName}`;
+}
+
+function insertHtml(html: string, collectionId: string) {
+	const placeholder = document.querySelector<HTMLElement>(
+		`[data-show-more-placeholder="${collectionId}"]`,
+	);
+	/**
+	 * it is expected that `html` will be a server-rendered instance
+	 * of the ExtraCardsContainer component
+	 */
+	if (placeholder) {
+		placeholder.insertAdjacentHTML('afterend', html);
+	}
+}
+
+function removeHtml(collectionId: string) {
+	const placeholder = document.querySelector<HTMLElement>(
+		`[data-show-more-placeholder="${collectionId}"]`,
+	);
+	if (placeholder) {
+		const nextSibling = placeholder.nextElementSibling;
+		if (nextSibling?.tagName === 'UL') nextSibling.remove();
+	}
+}
+
+export const ShowMore = ({
+	collectionId,
+	displayName,
+	editionId,
+}: {
+	collectionId: string;
+	displayName: string;
+	editionId: Edition;
+}) => {
+	const url = `https://api.nextgen.guardianapps.co.uk/${
+		editionId === 'INT' ? 'international' : editionId.toLowerCase()
```

Comment:
I think instead of editionId this part of the url is coming from `pressedPage.id` so we should pass this as a prop. For example, when we are here https://www.theguardian.com/uk/environment and click show more in Spotlight section:

![Screenshot 2022-06-22 at 17 06 10](https://user-images.githubusercontent.com/19683595/175080035-bc53479f-4921-49dc-be84-a01ab5b5f7e0.png)

Reply:
this issue should be fixed as of now 👍 

https://github.com/guardian/dotcom-rendering/blob/5e8ba23b89667911a8834f7773346532307c2146/dotcom-rendering/src/web/layouts/FrontLayout.tsx#L159-L162